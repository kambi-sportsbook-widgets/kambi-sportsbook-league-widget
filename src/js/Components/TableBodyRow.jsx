import React from 'react';

const TableBodyRow = ({ children }) => {
   return (
      <div
         className="kw-table-item KambiWidget-card-border-color KambiWidget-card-background-color--hoverable l-flexbox
               l-horizontal l-align-center l-pl-16 l-pr-16 l-pt-6 l-pb-6"
      >
         {children}
      </div>
   );
};

TableBodyRow.propTypes = {
   /**
    * Cell components
    */
   children: React.PropTypes.arrayOf(React.PropTypes.node).isRequired
};

export default TableBodyRow;
