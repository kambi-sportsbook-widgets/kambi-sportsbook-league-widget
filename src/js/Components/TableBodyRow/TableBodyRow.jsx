import React, { Children, PropTypes } from 'react';
import styles from './TableBodyRow.scss';

const TableBodyRow = ({ children }) => (
   <tr className={['KambiWidget-card-inner-border', styles.row].join(' ')}>
      {children}
      <td className='margin' />
   </tr>
);

TableBodyRow.propTypes = {
   /**
    * Inner rows
    */
   children: PropTypes.node
};

export default TableBodyRow;
