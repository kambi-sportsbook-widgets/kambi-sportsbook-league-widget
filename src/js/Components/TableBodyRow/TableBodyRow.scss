:local(.row) {
   height: 46px;

   td.margin {
      width: 16px;
   }
}
