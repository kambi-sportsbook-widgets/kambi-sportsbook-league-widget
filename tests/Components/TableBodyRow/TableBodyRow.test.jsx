/* eslint-env jest */
import React, { Children } from 'react';
import TableBodyRow from '../../../src/js/Components/TableBodyRow/TableBodyRow';
import ReactTestUtils from 'react-addons-test-utils';

let renderer;

describe('TableBodyRow DOM rendering', () => {
   beforeEach(() => {
      renderer = ReactTestUtils.createRenderer();
   });

   it('renders correctly without children', () => {
      expect(renderer.render(
         <TableBodyRow />
      )).toMatchSnapshot();
   });

   it('renders correctly with children', () => {
      expect(renderer.render(
         <TableBodyRow>
            <td/><td/>
         </TableBodyRow>
      )).toMatchSnapshot();
   });

});
