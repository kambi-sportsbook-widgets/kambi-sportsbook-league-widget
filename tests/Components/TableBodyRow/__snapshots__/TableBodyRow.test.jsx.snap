// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBodyRow DOM rendering renders correctly with children 1`] = `
<tr
  className="KambiWidget-card-inner-border row"
>
  <td />
  <td />
  <td
    className="margin"
  />
</tr>
`;

exports[`TableBodyRow DOM rendering renders correctly without children 1`] = `
<tr
  className="KambiWidget-card-inner-border row"
>
  <td
    className="margin"
  />
</tr>
`;
